demo/dismissable/dismissable-item.js

```diff
@@ -0,0 +1,218 @@
+  }
+
+  _dismiss() {
+    this.style.opacity = 0;
+
+    const collapseAnim = this.animate({
+      height: [ getComputedStyle(this).height, '0px']
+    }, {
+      duration: 100,
+      iterations: 1
+    });
+
+    collapseAnim.onfinish = this._fireRemove.bind(this);
+  }
+
+  _fireRemove() {
+    this.setPosition(0);
+
+    const event = new CustomEvent('remove', {
+      detail: { itemIndex: this.itemIndex },
+      bubbles: true
+    });
+    this.dispatchEvent(event);
+  }
+
+  settle(targetPosition) {
+    this.state = 'initial';
+    if (targetPosition === this.position) {
+      return;
+    }
+
+    const currentY = this.style.transform.split(',')[1];
+    const isDismiss = targetPosition !== 0;
+
+    const animation = this.animate({
+      transform: [
+        `translate(${this.position}px,${currentY}`,
+        `translate(${targetPosition}px,${currentY}`
+      ],
+      opacity: [ this.style.opacity, isDismiss ? 0 : 1 ]
+    }, {
+      duration: Math.abs(targetPosition - this.position) * 0.5,
+      iterations: 1
+    });
+
+    this.position = targetPosition;
+    animation.onfinish = () => isDismiss ? this._dismiss() : this.setPosition(0);
+  }
+
+  fling(velocityX) {
+    this.state = 'initial';
+    const targetPosition = velocityX < 0 ? -this.width : this.width;
+
+    const currentY = this.style.transform.split(',')[1];
+
+    const animation = this.animate({
+      transform: [
+        `translate(${this.position}px,${currentY}`,
+        `translate(${targetPosition}px,${currentY}`
+      ],
+      opacity: [ this.style.opacity, 0 ]
+    }, {
+      duration: Math.abs(targetPosition - this.position) / Math.abs(velocityX),
+      iterations: 1
+    });
+
+    animation.onfinish = this._dismiss.bind(this);
+  }
+
+  _settleToClosestPosition() {
+    const fraction = this.position / this.width;
+    if (fraction > 0.5) {
+      this.settle(this.width);
+    } else if (fraction < -0.5) {
+      this.settle(-this.width);
+    } else {
+      this.settle(0);
+    }
+  }
+
+  _onPointerDown(e, change) {
+    this.state = 'initial';
+    this.startX = change.clientX;
+    this.startY = change.clientY;
+    this.startPosition = 0;
+  }
+
+  _onPointerPan(e, change) {
+    if (this.state == 'initial') {
+      const deltaX = change.clientX - this.startX;
+      const deltaY = change.clientY - this.startY;
+
+      if (deltaX ** 2 + deltaY ** 2 < kTouchSlopValue ** 2) {
+        e.preventDefault();
+        return;
+      }
+
+      if (Math.abs(deltaY) > Math.abs(deltaX)) {
+        this._settleToClosestPosition();
+        return;
+      }
+
+      this.state = 'dragging';
+      this.width = this.offsetWidth;
+    }
+
+    if (this.state == 'dragging') {
+      this._tracker = this._tracker || new VelocityTracker();
+      this._tracker.update(e);
+
+      e.preventDefault();
+      const deltaX = change.clientX - this.startX;
+      this.position = this.startPosition + deltaX;
```

Comment:
nit: this should be done in `setPosition()` rather than here

Reply:
You mean the last line? like `setPosition(this.startPosition + deltaX)` ?

Reply:
Fixed